src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -171,14 +180,12 @@ async Task InnerProcessMessages()
 
         Task ReceiveMessage()
         {
-            return TaskEx.Run(async state =>
+            return Task.Run(async () =>
             {
-                var messagePump = (MessagePump)state;
-
                 try
                 {
-                    await messagePump.receiveStrategy.ReceiveMessage().ConfigureAwait(false);
-                    messagePump.receiveCircuitBreaker.Success();
+                    await receiveStrategy.ReceiveMessage(messageProcessingCancellationTokenSource.Token).ConfigureAwait(false);
+                    receiveCircuitBreaker.Success();
                 }
                 catch (OperationCanceledException)
```

Comment:
I noticed that David added an additional condition to these catches (https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/275/files#diff-bf785c1b37e39481e89856ea4195ca5c7e21f862b8eb2ffc43b63208a3c582e5R263). Would these make sense here as well?

Reply:
They are in the receive strategies, eg: https://github.com/Particular/NServiceBus.Transport.Msmq/pull/225/files#diff-4d6fd4708d86a05a95625c19f6da55d9d68be74498996e6a2aab8f90879a3a33R34 so I think that is already covered?